### PR TITLE
ResourceIdentifier exceptions are SafeLoggable for better, safer observability

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
ResourceIdentifier exceptions are SafeLoggable for better, safer observability
==COMMIT_MSG==

### CODE REVIEW

File: src/main/java/com/palantir/ri/ResourceIdentifier.java [hunk 1]
```diff
@@ -328,7 +332,7 @@ public static ResourceIdentifier valueOf(String rid) {
     public static ResourceIdentifier of(String rid) {
         ResourceIdentifier resultRid = tryOf(rid);
         if (resultRid == null) {
-            throw new IllegalArgumentException("Illegal resource identifier format: " + rid);
+            throw new SafeIllegalArgumentException("Illegal resource identifier format", UnsafeArg.of("rid", rid));
```
Comment (mpritham) [inline src/main/java/com/palantir/ri/ResourceIdentifier.java:335]: nit: This isn't incorrect now, but shall we also update the javadoc to `@throws SafeIllegalArgumentException`?

**Review by mpritham — APPROVED** [review]
Removed `merge-when-ready`. Approving.

File: src/main/java/com/palantir/ri/ResourceIdentifier.java (hunk 1 above)
Comment (carterkozak) [reply src/main/java/com/palantir/ri/ResourceIdentifier.java:335]: In general we don’t really want folks to catch the safe variants, so we don’t want folks to rely on the safe variant being api, but that would be more specific in javadoc. I could go either way :-)

Comment (mpritham) [reply src/main/java/com/palantir/ri/ResourceIdentifier.java:335]: Ok. If people shouldn't be catching _only_ the safe variants, then having the API specify the superclass lets them know what to catch instead.

No strong stance either, but leaning towards keeping as-is with the principle "the @ throws block in the javadoc should specify what devs to should catch".

Comment (mpritham) [reply src/main/java/com/palantir/ri/ResourceIdentifier.java:335]: Going to merge given this was the only comment.

**Pull request merged**